packages/desktop-client/src/components/budget/MobileBudget.tsx

```diff
@@ -0,0 +1,481 @@
+import React, { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useSpreadsheet } from 'loot-core/src/client/SpreadsheetProvider';
+import { type PrefsState } from 'loot-core/src/client/state-types/prefs';
+import { send, listen } from 'loot-core/src/platform/client/fetch';
+import * as monthUtils from 'loot-core/src/shared/months';
+import {
+  type CategoryEntity,
+  type CategoryGroupEntity,
+} from 'loot-core/src/types/models';
+
+import { type BoundActions, useActions } from '../../hooks/useActions';
+import { useCategories } from '../../hooks/useCategories';
+import { useSetThemeColor } from '../../hooks/useSetThemeColor';
+import { AnimatedLoading } from '../../icons/AnimatedLoading';
+import { theme } from '../../style';
+import { View } from '../common/View';
+import { SyncRefresh } from '../SyncRefresh';
+
+import { BudgetTable } from './MobileBudgetTable';
+import { prewarmMonth, switchBudgetType } from './util';
+
+const CATEGORY_BUDGET_EDIT_ACTION = 'category-budget';
+const BALANCE_MENU_OPEN_ACTION = 'balance-menu';
+
+type BudgetInnerProps = {
+  categories: CategoryEntity[];
+  categoryGroups: CategoryGroupEntity[];
+  prefs: PrefsState['local'];
+  loadPrefs: BoundActions['loadPrefs'];
+  savePrefs: BoundActions['savePrefs'];
+  budgetType: 'rollover' | 'report';
+  spreadsheet: ReturnType<typeof useSpreadsheet>;
+  applyBudgetAction: BoundActions['applyBudgetAction'];
+  collapseModals: BoundActions['collapseModals'];
+  pushModal: BoundActions['pushModal'];
+  getCategories: BoundActions['getCategories'];
+  createCategory: BoundActions['createCategory'];
+  updateCategory: BoundActions['updateCategory'];
+  deleteCategory: BoundActions['deleteCategory'];
+  moveCategory: BoundActions['moveCategory'];
+  createGroup: BoundActions['createGroup'];
+  updateGroup: BoundActions['updateGroup'];
+  deleteGroup: BoundActions['deleteGroup'];
+  moveCategoryGroup: BoundActions['moveCategoryGroup'];
+  sync: BoundActions['sync'];
+};
+
+function BudgetInner(props: BudgetInnerProps) {
+  const {
+    categoryGroups,
+    categories,
+    prefs,
+    loadPrefs,
+    savePrefs,
+    budgetType,
+    spreadsheet,
+    applyBudgetAction,
+    collapseModals,
+    pushModal,
+    createGroup,
+    updateGroup,
+    deleteGroup,
+    moveCategoryGroup,
+    createCategory,
+    updateCategory,
+    deleteCategory,
+    moveCategory,
+  } = props;
+
+  const currMonth = monthUtils.currentMonth();
+
+  const [bounds, setBounds] = useState({ start: currMonth, end: currMonth });
+  const [currentMonth, setCurrentMonth] = useState(currMonth);
+  const [initialized, setInitialized] = useState(false);
+  const [editMode, setEditMode] = useState(false);
+  const [editingBudgetCategoryId, setEditingBudgetCategoryId] = useState(null);
+  const [openBalanceActionMenuId, setOpenBalanceActionMenuId] = useState(null);
+
+  const loadCategories = async () => {
+    await props.getCategories();
+  };
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏no need to create a new function here. We can just use `props.getCategories` directly. Extra bonus: it maintains referential stability thus _should_ mean less re-renders.